File: java/src/test/java/problems1/SwapNodesInPairsTest.java
package problems1;

import concepts.SinglyListNode;
import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SwapNodesInPairsTest extends SampledTest {
    @Override
    public String getFileName() {
        return "swap_nodes_in_pairs.json";
    }

    @Test
    public void test() {
        for (SwapNodesInPairs solution : SwapNodesInPairs.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.swapPairs(SinglyListNode.of(sample.input.head)))
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, int[]> {
        public static class Input {
            public int[] head;
        }
    }
}

File: typescript/__tests__/problems1_100/swap-nodes-in-pairs.test.ts
import {deepEqual} from 'assert';
import SinglyListNode from '../../src/concepts/singly-list-node';
import SOLUTIONS from '../../src/problems1_100/swap-nodes-in-pairs';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    head: number[],
  },
  output: number[],
  message: string,
}

describe(
    'SwapNodesInPairs',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        deepEqual(
                            solution.swapPairs(SinglyListNode.of(...sample.input.head)),
                            SinglyListNode.of(...sample.output),
                            sample.message,
                        ),
                    ),
                ),
        ),
);
